src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -90,10 +93,12 @@
   private final FirebaseOptions options;
   private final TokenRefresher tokenRefresher;
   private final ThreadManager threadManager;
+  private final ThreadManager.FirebaseExecutor executor;
 
   private final AtomicBoolean deleted = new AtomicBoolean();
   private final Map<String, FirebaseService> services = new HashMap<>();
-  private final ListeningScheduledExecutorService executor;
+
+  private volatile ScheduledExecutorService scheduledExecutor;
```

Comment:
It seems like 'ScheduledExecutorService' doesn't spawn its threads until it gets used for the first time. If you were to initialize it here, the code is this class could be drastically simplified.

Reply:
This will initialize the `ThreadFactory` though (`ThreadManager.getThreadFactory()`) will get called. On App Engine this attempts to start a no-op background thread to see if the background thread support is available.Probably not a big deal, but just wanted to get your opinion on it before I make the change.

Reply:
I went through great lengths to make sure that we don't kick off random threads in the first place, so we should probably keep this as is. Thanks!